### PR TITLE
Stop double counting cancelled prints and move database location

### PR DESCRIPTION
**Pull request by Menace-Dennis**
I'm still getting used to GitHub, I didn't realise a pull request would keep updating with changes to the branch until approved, so creating a new pull request hahaha

When a print is cancelled, in stats this is counted as both "Cancelled" and "Failed". It should be one or the other really. Personally, in my fork, I chose to save the "Failed" status for prints that stopped because of errors. I then moved the "Error" category to the connection stack as this included errors that didn't also stop a print. This way the "Complete" + "Cancelled" + "Failed" is the same height as started.

Additionally, moved the database location to the data folder so that its included in backups.

Tested on Python3, OctoPrint 14.2

### CODE REVIEW

**Comment by AlexVerrico** [comment]
Hi @Menace-Dennis 
Have you added any checks/methods to migrate the database from the old location?
Cheers, Alex.

File: octoprint_stats/__init__.py [hunk 1]
```diff
@@ -20,10 +20,13 @@
...
     def __init__(self, plugin):
         self.DB_NAME_V1 = plugin._settings.global_get_basefolder("logs") + "/octoprint_stats.db"
-        self.DB_NAME = plugin._settings.global_get_basefolder("logs") + "/octoprint_stats.json"
+        self.DB_NAME_V2 = plugin._settings.global_get_basefolder("logs") + "/octoprint_stats.json"
+        self.DB_NAME = plugin._settings.global_get_basefolder("data") + "/octoprint_stats.json"
         
         if os.path.exists(self.DB_NAME_V1) == True:
             self.migrate_v1()
+        if os.path.exists(self.DB_NAME_V2) == True:
+            os.rename(self.DB_NAME_V2, self.DB_NAME)
```
Comment (Menace-Dennis) [inline octoprint_stats/__init__.py:29]: Moves the old database

**Comment by Menace-Dennis** [comment]
Hi @AlexVerrico,
Yep, I've highlighted the lines that do that :)
Cheers

**Comment by AlexVerrico** [comment]
Hi @Menace-Dennis 
Great, I do plan to merge these changes, but I'll have to do some checks on python 2 first.
It may take a week or two, but it will happen 😉 
Cheers, Alex.

**Comment by Menace-Dennis** [comment]
Haha, fair enough

**Pull request merged**